### PR TITLE
Feat/lightbox [Delivers #103225040]

### PR DESCRIPTION
**Pull request by chiefGui**
@fosterdill @jonschreiber @evandrodutra @harby

### CODE REVIEW

File: src/components/live-displays/Wall/Wall.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import './Wall.scss'
+
+export default React.createClass({
+  propTypes: {
+    assets: PropTypes.array.isRequired
+  },
+
+  getInitialState() {
+    return { assetLightbox: false }
```
Comment (fosterdill) [inline src/components/live-displays/Wall/Wall.jsx]: Might just want to make it 'hasAssetLightbox' so its obvious its a bool

**Comment by fosterdill** [comment]
Can we just change it to use 'React.createClass' then :+1:

File: src/components/live-displays/Wall/Wall.jsx (hunk 1 above)
Comment (chiefGui) [inline src/components/live-displays/Wall/Wall.jsx]: Perhaps I am doing wrong because it turns into a component when its state is changed. Like, false while nothing is set; otherwise, it is a component. What do you think?

File: src/components/live-displays/Wall/Wall.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      onClose={this._closeLightbox} />
+  },
+
+  renderGrid() {
+    return <Grid
+      assets={this.props.assets}
+      openLightbox={this._openLightbox} />
+  },
+
+  _openLightbox(asset) {
```
Comment (KasaiMagi) [inline src/components/live-displays/Wall/Wall.jsx]: no reason to have underscore prefix, is there?

Comment (chiefGui) [inline src/components/live-displays/Wall/Wall.jsx]: Technically, there's no reason. Just using it as a private method convention. Do you see any caveat on using that?

File: src/components/live-displays/Wall/Wall.jsx (hunk 1 above)
Comment (KasaiMagi) [inline src/components/live-displays/Wall/Wall.jsx]: Is this kept in state because there isnt a state management layer for this library?

Comment (chiefGui) [inline src/components/live-displays/Wall/Wall.jsx]: Yes, that's right.

File: src/components/live-displays/Wall/Wall.jsx (hunk 2 above)
Comment (KasaiMagi) [inline src/components/live-displays/Wall/Wall.jsx]: No problem with that if you keep consistent, but technically `renderLightbox` and `renderGrid` are considered private as well.

Comment (chiefGui) [inline src/components/live-displays/Wall/Wall.jsx]: You are right.

**Comment by chiefGui** [comment]
Eureka! Please, @fosterdill and @jonschreiber do a review because I did a few more updates. Thanks!

File: src/components/live-displays/Wall/Wall.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  },
+
+  renderGrid() {
+    return <Grid
+      assets={this.props.assets}
+      openLightbox={this.openLightbox} />
+  },
+
+  openLightbox(asset) {
+    this.setState({ hasAssetLightbox: asset })
```
Comment (petrbela) [inline src/components/live-displays/Wall/Wall.jsx]: better just set it to `true` so it's always boolean type

Comment (chiefGui) [inline src/components/live-displays/Wall/Wall.jsx]: I can't do that because asset, when set, isn't a boolean, but the asset itself in the type of an array. If you have a better suggestion to manage this situation, feel free to indicate. I thought to do this way to prevent to create another state for the component.

File: src/components/live-displays/Wall/Grid.jsx [hunk 4]
```diff
@@ -1,35 +1,40 @@
...
+    this.props.assets[0].tweet_only = true
+    this.props.assets[2].size = 'biggest'
 
-    return (
-      <div className="grid">
-        { this.props.assets.map(asset => this.renderAsset(asset)) }
-      </div>
-    )
+    this.props.assets[5].is_captionated = true
+    this.props.assets[6].size = 'biggest'
```
Comment (petrbela) [inline src/components/live-displays/Wall/Grid.jsx]: Shouldn't this configuration be in some test or demo class?

File: src/components/live-displays/Wall/Lightbox.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    let style = { backgroundImage: `url('${this.props.asset.url}')` }
+    return <div className="lightbox__hero" style={style}></div>
+  },
+
+  renderSidebar() {
+    return (
+      <aside className="lightbox__sidebar">
+        <div className="lightbox__details">
+          <section className="[ lightbox__section lightbox__user ]">
+            <img src={`//beta.avatars.io/instagram/${this.props.asset.user.username}`} className="lightbox__avatar" />
```
Comment (petrbela) [inline src/components/live-displays/Wall/Lightbox.jsx]: always prepend `https:`

File: src/components/live-displays/Wall/Wall.jsx (hunk 3 above)
Comment (petrbela) [inline src/components/live-displays/Wall/Wall.jsx]: I'm not sure I understand. `hasAssetLightbox` should either be a boolean or an asset object but not both at the same time. `getInitialState` sets `hasAssetLightbox: false` so it's a boolean. We shouldn't assign an object to the same attribute, it makes it hard to follow.

Comment (petrbela) [inline src/components/live-displays/Wall/Wall.jsx]: On the other hand, if you want to have asset object, rename the attribute to `lightboxAsset`, and initialize it with `null`, not `false`.

File: src/components/live-displays/Wall/Grid.jsx (hunk 4 above)
Comment (chiefGui) [inline src/components/live-displays/Wall/Grid.jsx]: That's what I am working on.

File: src/Menu.scss [hunk 6]
```diff
@@ -3,7 +3,7 @@
   top:  50%;
   left: 50%;
   transform: translate(-50%, -50%);
-  z-index: 999;
+  z-index: 999999;
```
Comment (harby) [inline src/Menu.scss]: Be aware with the z-index!

File: src/components/live-displays/Header/Header.jsx [hunk 7]
```diff
@@ -1,17 +1,18 @@
-import React from 'react'
+'use strict'
```
Comment (harby) [inline src/components/live-displays/Header/Header.jsx]: Why are you adding use strict back? I thought we tried to get rid of them, since you use babel.

File: src/components/live-displays/Wall/Lightbox.scss [hunk 8]
```diff
@@ -0,0 +1,103 @@
+.lightbox__ghost {
```
Comment (harby) [inline src/components/live-displays/Wall/Lightbox.scss]: Will it be possible to keep the class indentation based on it's presence in the DOM as well as to keep the properties in the order described in the CSS Style Guide? https://github.com/chute/css-styleguide#rules-order

File: src/components/shared/ProgressBar.jsx [hunk 9]
```diff
@@ -1,28 +1,29 @@
...
-
   getInitialState() {
     return {
-      timePassed: 0
+      interval: 5000,
+      stepDuration: 1000
     }
   },
 
+  getDefaultProps: () => ({
```
Comment (jonathan-potter) [inline src/components/shared/ProgressBar.jsx]: You shouldn't be using an arrow function here. None of the other functions are defined this way and you aren't using `this` anywhere, so there isn't a reason to use an arrow function at all.

File: src/components/shared/ProgressBar.jsx [hunk 10]
```diff
@@ -1,28 +1,29 @@
...
 
-  getDefaultProps: () => ({
-    interval: 5000,
-    stepDuration: 1000
-  }),
-
   getInitialState() {
     return {
-      timePassed: 0
+      interval: 5000,
```
Comment (jonathan-potter) [inline src/components/shared/ProgressBar.jsx]: neither `interval` or `stepDuration` are ever used with state in this component. This looks like a mistake.

File: src/components/shared/Timer.jsx [hunk 11]
```diff
@@ -1,15 +1,21 @@
-import React from 'react'
+'use strict'
+
+import React, { Component, PropTypes } from 'react'
```
Comment (jonathan-potter) [inline src/components/shared/Timer.jsx]: you added `Component` to this file, but it isn't used.

Comment (jonathan-potter) [inline src/components/shared/Timer.jsx]: this is true in at least one other file.

File: src/components/shared/ProgressBar.jsx (hunk 9 above)
Comment (jonathan-potter) [inline src/components/shared/ProgressBar.jsx]: i guess you just moved this, so whatever

File: src/components/live-displays/Wall/Wall.jsx (hunk 3 above)
Comment (jonathan-potter) [inline src/components/live-displays/Wall/Wall.jsx]: so what @petr is getting at here is that the name `hasAssetLightbox` sounds like a boolean, but it really isn't treated like a boolean anywhere.

it looks like it is an `asset` variable for the `lightbox`, so the name should show that it is an `asset`, not a boolean. and it should never be a boolean.

`lightboxedAsset`, `currentAsset`, `displayedAsset`... something that sounds like what it is would be much better than `hasAssetLightbox`

File: src/components/live-displays/ThreeAssets/ThirdAsset.jsx [hunk 12]
```diff
@@ -38,21 +38,22 @@ export default React.createClass({
...
+      <div className="s-author__username">
+        <p>{this.props.asset.user.username}</p>
       </div>
     )
   },
 
-  render() {
-    let style = { backgroundImage: `url("${this.props.asset.url}")` }
+  renderServiceBadge() {
+    let className = classNames('fa', `fa-${this.props.asset.service}`)
```
Comment (jonathan-potter) [inline src/components/live-displays/ThreeAssets/ThirdAsset.jsx]: sooo.... the classname `fa`... I'm not sure where this started or what it's for.

Is there a really good reason to have unreadable classes? @harby @chiefGui
